Samples~/LootLockerExamples/8 - Virtual Currency Purchases/LootLockerVirtualStore.cs

```diff
@@ -0,0 +1,126 @@
+
+            LootLockerSettingsOverrider.OverrideSettings();
+
+            LootLockerSDKManager.StartGuestSession((response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not start guest session");
+                    return;
+                }
+                FetchPlayerWallet(response.player_ulid);
+                FetchVirtualProductsFromCatalog();
+
+            });
+        }
+
+        public void FetchPlayerWallet(string ulid)
+        {
+            LootLockerSDKManager.GetWalletByHolderId(ulid, LootLockerEnums.LootLockerWalletHolderTypes.player, (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not start guest session");
+                    return;
+                }
+
+                walletID = response.id;
+                GetPlayerGold();
+                GiveGold();
+            });
+        }
+
+        public void GetPlayerGold()
+        {
+            LootLockerSDKManager.ListBalancesInWallet(walletID, (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not list balances");
+                    return;
+                }
+
+                currencyCode = response.balances[0].currency.code.ToUpper() + ": ";
+
+                goldAmount.text = currencyCode + response.balances[0].amount;
+            });
+        }
+
+
+        public void FetchVirtualProductsFromCatalog()
+        {
+            LootLockerSDKManager.ListCatalogItems(catalogKey, 0, "", (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not fetch Catalog Items");
+                    return;
+                }
+
+
+                foreach (var item in response.entries)
+                {
+
+                    switch(item.entity_kind){
+                        case LootLockerEnums.LootLockerCatalogEntryEntityKind.asset:
+                            var something = response.asset_details[item.GetItemDetailsKey()];
+                            Debug.Log(something.name);
+                            break;
+
+                    }
+
+
+
+                   
+                    Debug.Log("Item: " + item.entity_name + " | " + item.prices[0].display_amount);
+                    var obj = Instantiate(virtualPurchaseProductPrefab, storeUI);
+                    obj.GetComponent<VirtualPurchaseProduct>().CreateProduct(item.entity_name,
+                                                                                item.prices[0].amount,
+                                                                                   item.catalog_listing_id);
+                }
+            });
+        }
+
+        public void GiveGold()
+        {
+            LootLockerSDKManager.CreditBalanceToWallet(walletID, "01J3MK8BTC6JC9850YTXGXP1H5", "200", (response) =>
```

Comment:
What is this Id? Is it the currency ID? Can we please list currencies or something instead and get this id dynamically? It's hard I think for our users to parse that this is what you're doing behind the scenes.